pennylane/labs/resource_estimation/ops/op_math/symbolic.py

```diff
@@ -178,3 +182,114 @@ def pow_resource_decomp(
     def tracking_name(base_class, z, base_params) -> str:
         base_name = base_class.tracking_name(**base_params)
         return f"Pow({base_name}, {z})"
+
+
+class ResourceExp(Exp, re.ResourceOperator):
+    """Resource class for Exp"""
+
+    @staticmethod
+    def _resource_decomp(base_class, coeff, num_steps, **kwargs):
+
+        while isinstance(base_class, SProd):
+            coeff *= base_class.scalar
+            base_class = base_class.base
+
+        # Custom exponential operator resources:
+        if isinstance(base_class, re.ResourceOperator):
+            try:
+                return base_class.exp_resource_decomp(coeff, num_steps, **kwargs)
+            except re.ResourcesNotDefined:
+                pass
+
+        # PauliRot resource decomp:
+        if (pauli_sentence := base_class.pauli_rep) and math.real(coeff) == 0:
+            if qml.pauli.is_pauli_word(base_class):
+                num_wires = len(base_class.wires)
```

Comment:
Same here, `num_wires` should be provided in the minimal set of information needed to compute a resource estimate.